src/coreclr/src/pal/tests/palsuite/paltests.cpp

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*=============================================================
+**
+** Source: paltests.cpp
+**
+** Purpose: Entrypoint for all the pal tests. Written to avoid any
+**          standard library usage
+**
+**============================================================*/
+#include <palsuite.h>
+
+PALTest* PALTest::s_tests = 0;
+
+int __cdecl main(int argc, char *argv[])
+{
+    if (argc < 2)
+        return FAIL;
+    
+    PALTest *testCur = PALTest::s_tests;
+    for (;testCur != 0; testCur = testCur->_next)
+    {
+        int i = 0;
+        bool stringMatches = false;
+        while (testCur->_name[i] == argv[1][i])
+        {
+            if (testCur->_name[i] == '\0')
+            {
+                stringMatches = true;
+                break;
+            }
+            i++;
+        }
+        if (!stringMatches)
+            continue;
+
+        for (int i = 1; i < (argc - 1); i++)
```

Comment:
Unfortunately, no, that doesn't work. The CreateProcess tests relaunch the current process, and they determine the appropriate binary by examining argv[0], thus the need to shift instead of passing argv + 1.

Reply:
Ah, ok.